gtdynamics/factors/JointMeasurementFactor.h

```diff
@@ -92,5 +95,21 @@ class JointMeasurementFactor
     gtsam::traits<double>::Print(measured_joint_coordinate_, "  measured: ");
     this->noiseModel_->print("  noise model: ");
   }
+
+  bool equals(const gtsam::NonlinearFactor& other,
+              double tol = 1e-9) const override {
+    const This* e = dynamic_cast<const This*>(&other);
+    return e != nullptr && Base::equals(*e, tol) &&
```

Comment:
Don't need `e != nullptr` here, just use `e && ...`

Reply:
True but it's nice to be explicit. :)